src/main/java/uk/ac/ebi/ega/egafuse/runner/CommandLineOptionParser.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.egafuse.runner;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.internal.Strings;
+import joptsimple.util.PathConverter;
+import uk.ac.ebi.ega.egafuse.model.Input;
+
+public class CommandLineOptionParser {
+    private static final String OPTIONS_HELP = "h";
+
+    public static Input parser(String[] args) throws IOException {
+        final OptionParser optionParser = buildParser();
+        final OptionSet optionSet = optionParser.parse(args);
+        if (optionSet.has(OPTIONS_HELP)) {
+            optionParser.printHelpOn(System.out);
+            System.exit(0);
+        }
+
+        final Input input = new Input();
+        input.setConnection(Integer.valueOf(optionSet.valueOf("c").toString()));
+
+        if (optionSet.has("cf")) {
+            Path credFilePath = (Path) optionSet.valueOf("cf");
+            Map<String, String> credentials = readAndValidateCredentialFile(credFilePath);
+            input.setUsername(credentials.get("username"));
+            input.setPassword(credentials.get("password"));
+            input.setCredFile(credFilePath);
```

Comment:
If the credentials is only used to give username and password to the "Input" you don't need to store it anywhere else as the information has already been colected. Besides that the credFile variable is no longer needed if the credential file only holds user/password values.

Reply:
Done